### PR TITLE
Added ConvertString2Int project, customConvert method and tests.

### PR DESCRIPTION
**Pull request by ZokiDespotovski**

### CODE REVIEW

File: numbers-csharp/ConvertString2Int/ConvertString2Int/Benchmark.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        static string BenchmarkIntParse(string strValue, int reps)
+        {
+            int number;
+            var watch = new Stopwatch();
+            watch.Start();
+            for (int i = 0; i < reps; i++)
+            {
+                number = int.Parse(strValue);
+            }
+            watch.Stop();
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2Int/Benchmark.cs:24]: You'll need to remove the StopWatch after the suggestions in the article, so read them first.

File: numbers-csharp/ConvertString2Int/ConvertString2Int/CustomConvert.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public static class CustomConvert
+    {
+        public static int Parse(string strVal)
+        {
+            int num = 0;
+            for (int i = 0; i < strVal.Length; i++)
+            {
+                num = num * 10 + (strVal[i] - '0');
+            }
+            return num;
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2Int/CustomConvert.cs:18]: Newlines after the variable definition and before the return statement. use var instead of int for variable definition.

File: numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace ConvertString2IntTests
+{
+    [TestClass]
+    public class ConvertString2IntTests
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs:7]: As I've mentioned in the article, try to create some tests that don't cover just the "happy path". It's important to test for exceptions as well. You'll see what I mean if you read the article review first. Cover empty strings and alphanumeric strings as well. It's a small article and it's easy to cover these edge cases.

File: numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            Assert.AreEqual(number, numberConvertedFromString);
+        }
+
+        [TestMethod]
+        public void GivenString_WhenConvertedWithIntTryParse_ThenReturnInt()
+        {
+            var number = 3;
+            var stringValue = "3";
+
+            int numberConvertedFromString = int.TryParse(stringValue, out numberConvertedFromString) ? numberConvertedFromString : 0;
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs:26]: Change as per article suggestion.

File: numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            Assert.AreEqual(number, numberConvertedFromString);
+        }
+
+        [TestMethod]
+        public void GivenString_WhenConvertedWithCustomConvert_ThenReturnInt()
+        {
+            var number = 3;
+            var stringValue = "3";
+
+            int numberConvertedFromString = 0;
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs:48]: Use vars please.

File: numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            for (int i = 0; i < stringValue.Length; i++)
+            {
+                numberConvertedFromString = numberConvertedFromString * 10 + (stringValue[i] - '0');
+            }
+
+            Assert.AreEqual(number, numberConvertedFromString);
+        }
+    }
+
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs:57]: Empty line.

**Review by CodeMazeBlog — COMMENTED** [review]
There are some small changes, but you'll also have to do some rewriting when you read the article suggestions.

File: numbers-csharp/ConvertString2Int/ConvertString2Int/CustomConvert.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace ConvertString2Int
+{
+    public static class CustomConvert
+    {
+        public static int Parse(string strVal)
+        {
+            var num = 0;
+
+            for (int i = 0; i < strVal.Length; i++)
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2Int/CustomConvert.cs:15]: var

File: numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        [TestMethod]
+        public void GivenString3_WhenConvertedWithIntTryParse_ThenReturnInt()
+        {
+            var number = 3;
+            var stringValue = "3";
+
+            var numberConvertedFromString = 0;
+            int.TryParse(stringValue, out numberConvertedFromString);
+
+            Assert.AreEqual(number, numberConvertedFromString);
```
Comment (CodeMazeBlog) [inline numbers-csharp/ConvertString2Int/ConvertString2IntTests/ConvertString2IntTests.cs:40]: Separate Arrange/Act/Assert with newlines. Applies to other methods as well.
